Source/Core/DiscIO/VolumeVerifier.cpp

```diff
@@ -831,6 +820,27 @@ void VolumeVerifier::Finish()
     }
   }
 
+  if (IsDisc(m_volume.GetVolumeType()) &&
+      (m_volume.IsSizeAccurate() || m_volume.SupportsIntegrityCheck()))
+  {
+    u64 volume_size = m_volume.IsSizeAccurate() ? m_volume.GetSize() : m_biggest_verified_offset;
+    if (m_biggest_referenced_offset > volume_size)
+    {
+      const bool second_layer_missing =
+          m_biggest_referenced_offset > SL_DVD_SIZE && m_volume.GetSize() >= SL_DVD_SIZE;
```

Comment:
I'm probably misreading this, but shouldn't the second check be <=?

Reply:
This is intentional but probably a bit confusing. `second_layer_missing` should be `false` if `m_volume.GetSize() < SL_DVD_SIZE`, so that the user always will get the message about needing to merge the split files if they have split files (regardless of whether the disc is supposed to be dual layer). As for why I used `m_volume.GetSize() >= SL_DVD_SIZE` instead of `m_volume.GetSize() == SL_DVD_SIZE`, it's because I also want to count disc images of the size `SL_DVD_R_SIZE` as single layer disc images. I could change this to `m_volume.GetSize() >= SL_DVD_SIZE && m_volume.GetSize() <= SL_DVD_R_SIZE` to make it make more sense, but when I originally wrote the code, I figured it didn't matter since there's no reasonable way that you'd get a file that's bigger than a single layer disc but smaller than a two layer disc. (The problem with split files is limited to 4 GiB, after all.)